src/op/declare/collective_comm.cc

```diff
@@ -251,6 +251,29 @@ RAF_OP_DECLARE("raf.op._recv", Recv)
     .set_attr<TOpPattern>("TOpPattern", kOpaque)
     .set_attr<TRAFCollective>("TRAFCollective", true);
 
+void Gather(const CallValues& call) {
+  const auto* args = call->args.as<CommGatherArgs>();
+  CHECK(args != nullptr);
+  const DLTensor* x = args->x[0];
+  size_t size = GetGlobalCommunicator()->size;
+  std::vector<int64_t> shape(x->shape, x->shape + x->ndim);
+  shape[0] = shape[0] * size;
+  call->device = x->device;
+  if (GetGlobalCommunicator()->rank == args->root) {
+    call->out = TensorValue::Assemble(/*ctx=*/x->device,
+                                      /*dtype=*/x->dtype,
+                                      /*shape=*/shape);
+  } else {
+    call->out = TensorValue::Assemble(/*ctx=*/x->device,
+                                      /*dtype=*/x->dtype,
+                                      /*shape=*/std::vector<int64_t>{});
```

Comment:
- The type infer function seems missing (see https://github.com/awslabs/raf/blob/main/src/op/ty/collective_comm.cc).
- I'm not sure if this can pass type checking or not as it depends on the use cases.